librz/core/cmd/cmd_debug.c

```diff
@@ -2235,51 +2199,220 @@ RZ_IPI RzCmdStatus rz_cmd_debug_step_until_optype_handler(RzCore *core, int argc
 	return RZ_CMD_STATUS_OK;
 }
 
+// dsue
 RZ_IPI RzCmdStatus rz_cmd_debug_step_until_esil_handler(RzCore *core, int argc, const char **argv) {
 	step_until_esil(core, argv[1]);
 	rz_core_dbg_follow_seek_register(core);
 	return RZ_CMD_STATUS_OK;
 }
 
+// dsuf
 RZ_IPI RzCmdStatus rz_cmd_debug_step_until_flag_handler(RzCore *core, int argc, const char **argv) {
 	step_until_flag(core, argv[1]);
 	rz_core_dbg_follow_seek_register(core);
 	return RZ_CMD_STATUS_OK;
 }
 
+// dt
+RZ_IPI RzCmdStatus rz_cmd_debug_trace_handler(RzCore *core, int argc, const char **argv) {
+	RzDebugTracepoint *t;
+	if ((t = rz_debug_trace_get(core->dbg, core->offset))) {
+		rz_cons_printf("offset = 0x%" PFMT64x "\n", t->addr);
+		rz_cons_printf("opsize = %d\n", t->size);
+		rz_cons_printf("times = %d\n", t->times);
+		rz_cons_printf("count = %d\n", t->count);
+		// TODO cons_printf("time = %d\n", t->tm);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dtl
+RZ_IPI RzCmdStatus rz_cmd_debug_traces_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	rz_debug_trace_list(core->dbg, state->mode, core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dtl=
+RZ_IPI RzCmdStatus rz_cmd_debug_traces_ascii_handler(RzCore *core, int argc, const char **argv) {
+	rz_debug_traces_ascii(core->dbg, core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dt+
+RZ_IPI RzCmdStatus rz_cmd_debug_trace_add_handler(RzCore *core, int argc, const char **argv) {
+	int count = argc > 2 ? rz_num_math(core->num, argv[1]) : 1;
+	RzAnalysisOp *op = rz_core_op_analysis(core, core->offset, RZ_ANALYSIS_OP_MASK_HINT);
+	if (!op) {
+		RZ_LOG_ERROR("Cannot analyze opcode at 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_OK;
+	}
+
+	RzDebugTracepoint *tp = rz_debug_trace_add(core->dbg, core->offset, op->size);
+	if (!tp) {
+		rz_analysis_op_free(op);
+		return RZ_CMD_STATUS_OK;
+	}
+	tp->count = count;
+	rz_analysis_trace_bb(core->analysis, core->offset);
+	rz_analysis_op_free(op);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dt++
+RZ_IPI RzCmdStatus rz_cmd_debug_trace_add_addrs_handler(RzCore *core, int argc, const char **argv) {
+	for (int i = 1; i < argc; ++i) {
+		ut64 addr = rz_num_get(NULL, argv[i]);
+		(void)rz_debug_trace_add(core->dbg, addr, 1);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dt-
+RZ_IPI RzCmdStatus rz_cmd_debug_traces_reset_handler(RzCore *core, int argc, const char **argv) {
+	rz_tree_reset(core->dbg->tree);
+	rz_debug_trace_free(core->dbg->trace);
+	rz_debug_tracenodes_reset(core->dbg);
+	core->dbg->trace = rz_debug_trace_new();
+	return RZ_CMD_STATUS_OK;
+}
+
+// dta
+RZ_IPI int rz_cmd_debug_trace_addr(void *data, const char *input) {
+	RzCore *core = (RzCore *)data;
+	rz_debug_trace_at(core->dbg, input);
+	return 0;
```

Comment:
`RZ_CMD_STATUS_OK`

Reply:
It's type is `RZ_CMD_DESC_TYPE_OLDINPUT`, and it return an `int`.
So it probably shouldn't return a `RzCmdStatus`.

Reply:
That's right